File: viralt-app/src/types/prize.ts
/** Prize Types — Types for the prize system */

/** Main prize information */
export interface PrizeInfo {
  prizeId: number;
  campaignId: number;
  title: string;
  description: string | null;
  image: string | null;
  quantity: number;
  prizeType: number;
  couponCode: string | null;
  sortOrder: number;
  minEntriesRequired: number | null;
}

/** Data required to create a new prize (no ID) */
export interface PrizeInsertInfo {
  campaignId: number;
  title: string;
  description: string | null;
  image: string | null;
  quantity: number;
  prizeType: number;
  couponCode: string | null;
  sortOrder: number;
  minEntriesRequired: number | null;
}

/** Data required to update an existing prize (includes ID) */
export interface PrizeUpdateInfo extends PrizeInsertInfo {
  prizeId: number;
}

/** Prize list API result */
export interface PrizeListResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
  prizes: PrizeInfo[];
}

/** Prize get API result */
export interface PrizeGetResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
  prize: PrizeInfo;
}

/** Status-only result */
export interface PrizeStatusResult {
  sucesso: boolean;
  mensagem: string;
  erros: string[] | null;
}

File: Viralt.Domain/Models/Prize.cs
namespace Viralt.Domain.Models;

public class Prize
{
    public long PrizeId { get; set; }
    public long CampaignId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string Image { get; set; }
    public int Quantity { get; set; }
    public int PrizeType { get; set; }
    public string CouponCode { get; set; }
    public int SortOrder { get; set; }
    public int? MinEntriesRequired { get; set; }

    public virtual Campaign Campaign { get; set; }
    public virtual ICollection<Winner> Winners { get; set; } = new List<Winner>();
}
